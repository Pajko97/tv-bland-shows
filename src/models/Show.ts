interface SingleShow {
    averageRuntime: number,
    dvdCountry: null
    ended: string,
    externals: Object
    genres: Array<any>,
    id: number,
    image: {
        medium : string,
        original : string
    },
    language: string
    name: string,
    network: {
        country : object,
        id: number,
        name : string
    },
    officialSite: null,
    premiered: string,
    rating: {
        average : any
    },
    runtime: number,
    schedule: {
        time : string,
        days: [string]
    },
    status: String,
    summary: string,
    type: string,
    updated: number,
    url: string,
    webChannel: null,
    weight: number,
    _links: Object
}   

export default SingleShow