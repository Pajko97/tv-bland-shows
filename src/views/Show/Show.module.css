.show_wrapper {
  padding: 10rem;
  font-family: "Red Hat Display";
}

.show_header {
  display: flex;
}
.show_header img {
  margin-right: 5rem;
  min-height: 100%;
  max-width: 25rem;
  max-height: auto;
}

.show_basic_info {
  max-width: 40rem;
}
.show_basic_info p {
  line-height: 150%;
}

.show_basic_info .rate {
  margin-left: 0.5rem;
}

.show_details {
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
}

@media only screen and (max-width: 800px) {
  .show_wrapper {
    padding: 3rem;
  }
  .show_basic_info {
    margin-top: 2rem;
  }
  .show_header {
    display: flex;
    flex-direction: column;
  }
  .show_header img {
    margin-right: 0;
  }
  .show_details {
    flex-direction: column;
  }
}
