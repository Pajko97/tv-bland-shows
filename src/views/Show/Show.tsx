import React, { useEffect, useState } from 'react';
import { useParams } from 'react-router';
import API from '../../api/shows-api';
import SingleShow from '../../models/Show'
import striptags from '../../helpers/striptags'
import Rating from 'react-rating'
import ActorList from "../../components/Show/ActorList/ActorList"
import style from './Show.module.css'
import ShowInfo from '../../components/Show/ShowInfo/ShowInfo';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faCoffee, faStar, faStarHalfStroke } from '@fortawesome/free-solid-svg-icons'

type Props = Object

function Show(props: Props) {
  const [show, setShow] = useState<SingleShow>()

  let { id }: any = useParams();
  let showId = parseInt(id)

  useEffect(() => {
    let api = new API();
    api.getShow(showId)
      .then(res => setShow(res.data))
  }, [])

  const calcRating = (rating: number) =>
	rating > 5 ? Math.round((rating / 2) * 10) / 10 : rating

  return (
    <React.Fragment>
      {show && <div className={style.show_wrapper}>
        <div>
          <div className={style.show_header}>
            <img src={show.image.medium} alt="show cover"></img>
            <div className={style.show_basic_info}>
            <Rating
              emptySymbol={<FontAwesomeIcon icon={faStarHalfStroke} />}
              readonly
              initialRating={calcRating(show.rating.average)}
              fullSymbol={<FontAwesomeIcon icon={faStar}/>}

            />
            <span className={style.rate}>{show.rating.average ? `${show.rating.average}/5` : 'Not provided'}</span>
            <h1>{show.name}</h1>
            <p>{striptags(show.summary)}</p>
            </div>
          </div>

          <div className={style.show_details}>
            <ShowInfo show={show} />
            <ActorList showId={showId} />
          </div>
        </div>
      </div>}
    </React.Fragment>
  );
}

export default Show

