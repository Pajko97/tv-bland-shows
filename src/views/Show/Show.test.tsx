import React from "react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Show from "../Show/Show";

it('renders Show', () => {
  let render;

    render = renderer.create(
        <BrowserRouter>
          <Show showId={6657}/>
        </BrowserRouter>
    ).toJSON();

expect(render).toMatchSnapshot();

});
