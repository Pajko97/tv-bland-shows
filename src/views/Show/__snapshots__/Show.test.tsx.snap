// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Show 1`] = `null`;
